rust/hyperlane-base/src/contract_sync/cursor.rs

```diff
@@ -44,7 +44,7 @@ where
     /// Wait based on how close we are to the tip and update the tip,
     /// i.e. the highest block we may scrape.
     async fn rate_limit(&mut self) -> ChainResult<()> {
-        let update_tip = self.last_tip_update.elapsed() >= Duration::from_secs(30);
+        let update_tip = self.last_tip_update.elapsed() >= Duration::from_secs(5);
```

Comment:
I just want to highlight that this is going to make a significant impact on the number of queries we are making.

Reply:
indeed
@asaj is working on something better here
I can revert if desired